### PR TITLE
feature/azure auth

### PR DESCRIPTION
**Pull request by duckth**
- add lots of sample stuff
- convert client go wasm
- remove sample
- add functional auth

### CODE REVIEW

File: NEWgIT.Client/NEWgIT.Client.csproj [hunk 1]
```diff
@@ -7,9 +7,15 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="7.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="7.0.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Http" Version="7.0.0" />
     <PackageReference Include="Radzen.Blazor" Version="4.3.6" />
-    <ProjectReference Include="..\NEWgIT.Core\NEWgIT.Core.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <_ContentIncludedByDefault Remove="wwwroot\sample-data\weather.json" />
```
Comment (duckth) [inline NEWgIT.Client/NEWgIT.Client.csproj:18]: slet

File: NEWgIT.Client/Pages/FetchData.razor [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+<h1>Weather forecast</h1>
+
+<p>This component demonstrates fetching data from the server.</p>
+
+@if (forks == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
```
Comment (duckth) [inline NEWgIT.Client/Pages/FetchData.razor:16]: slet hele siden

File: NEWgIT.Client/Pages/Index.razor [hunk 3]
```diff
@@ -254,4 +259,10 @@
         public DateOnly Date { get; set; }
         public int Commits { get; set; }
     }
+
+    public record AuthorsDTO(Dictionary<string, Dictionary<DateOnly, int>> AuthorFrequencies);
+    
+    public record ForksDTO(ICollection<string> Forks);
+
+    public record FrequenciesDTO(Dictionary<DateOnly, int> Frequencies);
```
Comment (duckth) [inline NEWgIT.Client/Pages/Index.razor:267]: for at undgå dependency på Newgit.Core - vi må lige overveje hvad der er det "rigtige" at gøre

File: NEWgIT.Client/Shared/NavMenu.razor [hunk 4]
```diff
@@ -1,17 +1,27 @@
 ﻿<div class="top-row ps-3 navbar navbar-dark">
     <div class="container-fluid">
-        <a class="navbar-brand" href="">NEWgIT</a>
+        <a class="navbar-brand" href="">BlazorSample</a>
```
Comment (duckth) [inline NEWgIT.Client/Shared/NavMenu.razor:3]: slet hele siden

Comment (duckth) [reply NEWgIT.Client/Shared/NavMenu.razor:3]: medmindre vi bruger den?

File: NEWgIT.Client/wwwroot/appsettings.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "AzureAdB2C": {
+    "Authority": "https://newgitb2c.b2clogin.com/newgitb2c.onmicrosoft.com/B2C_1_signupsignin",
+    "ClientId": "3c893c1f-4f13-4b01-ade1-442988ed6c23",
```
Comment (duckth) [inline NEWgIT.Client/wwwroot/appsettings.json:4]: burde det her være i secrets? hmm..

File: NEWgIT.Client/wwwroot/css/app.css [hunk 6]
```diff
@@ -34,6 +34,44 @@ a, .btn-link {
     color: red;
 }
```
Comment (duckth) [inline NEWgIT.Client/wwwroot/css/app.css:35]: Den her fil er fuldstændig kaos - jeg tror det virker though

File: NEWgIT.Client/wwwroot/css/open-iconic/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
+[Open Iconic v1.1.1](https://github.com/iconic/open-iconic)
```
Comment (duckth) [inline NEWgIT.Client/wwwroot/css/open-iconic/README.md:1]: nok bare slet

File: NEWgIT.Client/Pages/Counter.razor [hunk 8]
```diff
@@ -0,0 +1,18 @@
+﻿@page "/counter"
```
Comment (fredpetersen) [inline NEWgIT.Client/Pages/Counter.razor:1]: 🗡️ hele siden

File: NEWgIT.Client/Pages/FetchData.razor [hunk 9]
```diff
@@ -0,0 +1,53 @@
+﻿@page "/fetchdata"
```
Comment (fredpetersen) [inline NEWgIT.Client/Pages/FetchData.razor:1]: 🗡️ hele siden

File: NEWgIT.Client/Program.cs [hunk 10]
```diff
@@ -1,27 +1,29 @@
...
-var app = builder.Build();
-
-// Configure the HTTP request pipeline.
-if (!app.Environment.IsDevelopment())
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+using NEWgIT.Client;
+using System.Net.Http;
+using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+
+var api_scope = @"https://newgitb2c.onmicrosoft.com/newgit-api/access_as_user";
```
Comment (fredpetersen) [inline NEWgIT.Client/Program.cs:7]: Dette er vel kun auth_scope? Eller misforstår jeg?

File: NEWgIT.Client/wwwroot/css/open-iconic/README.md (hunk 7 above)
Comment (fredpetersen) [inline NEWgIT.Client/wwwroot/css/open-iconic/README.md:1]: Hvad er det her?
Kan det også bare 🗡️ ?

**Review by fredpetersen — CHANGES_REQUESTED** [review]

**Comment by fredpetersen** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=duckth_bdsa-project-newgit&pullRequest=87)
> 
> [Bug](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=SECURITY_HOTSPOT) [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=CODE_SMELL) [2 Code Smells](https://sonarcloud.io/project/issues?id=duckth_bdsa-project-newgit&pullRequest=87&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=duckth_bdsa-project-newgit&pullRequest=87&metric=coverage&view=list) No Coverage information [](https://sonarcloud.io/component_measures?id=duckth_bdsa-project-newgit&pullRequest=87&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=duckth_bdsa-project-newgit&pullRequest=87&metric=new_duplicated_lines_density&view=list)

Kan være du også lige skal have styr på det der security hotspot før den merges

File: NEWgIT.Client/wwwroot/css/open-iconic/README.md (hunk 7 above)
Comment (duckth) [reply NEWgIT.Client/wwwroot/css/open-iconic/README.md:1]: Jeg ved ikke hvad det er
Tror det er med som default?

**Review by lucasfth — APPROVED** [review]
looks good

**Pull request merged**